src/pages/RepoPage/CoverageTab/subroute/FileExplorer/FileListTable/FileListTable.tsx

```diff
@@ -0,0 +1,217 @@
+import {
+  createColumnHelper,
+  flexRender,
+  getCoreRowModel,
+  useReactTable,
+} from '@tanstack/react-table'
+import cs from 'classnames'
+import { useMemo, useState } from 'react'
+
+import { OrderingDirection } from 'services/repos'
+import { Row } from 'shared/ContentsTable/utils'
+import Icon from 'ui/Icon'
+
+import { useRepoBranchContentsTable } from '../hooks'
+import { Loader, RepoContentsResult } from '../shared'
+
+const columnHelper = createColumnHelper<Row>()
+
+function getOrderingDirection(sorting: Array<{ id: string; desc: boolean }>) {
+  const state = sorting.at(0)
+
+  if (state) {
+    const direction = state?.desc
+      ? OrderingDirection.DESC
+      : OrderingDirection.ASC
+
+    let ordering = undefined
+    if (state.id === 'name') {
+      ordering = 'NAME'
+    }
+
+    if (state.id === 'percentCovered') {
+      ordering = 'COVERAGE'
+    }
+
+    if (state.id === 'hits') {
+      ordering = 'HITS'
+    }
+
+    if (state.id === 'misses') {
+      ordering = 'MISSES'
+    }
+
+    if (state.id === 'partials') {
+      ordering = 'PARTIALS'
+    }
+
+    if (state.id === 'lines') {
+      ordering = 'LINES'
+    }
+
+    return { direction, ordering }
+  }
+
+  return undefined
+}
+
+export const getBaseColumns = () => {
+  const baseColumns = [
+    columnHelper.accessor('name', {
+      id: 'name',
+      header: () => 'Files',
+      cell: ({ renderValue }) => renderValue(),
+    }),
+    columnHelper.accessor('lines', {
+      id: 'lines',
+      header: () => 'Tracked lines',
+      cell: ({ renderValue }) => renderValue(),
+    }),
+    columnHelper.accessor('hits', {
+      id: 'hits',
+      header: () => 'Covered',
+      cell: ({ renderValue }) => renderValue(),
+    }),
+    columnHelper.accessor('partials', {
+      id: 'partials',
+      header: () => 'Partial',
+      cell: ({ renderValue }) => renderValue(),
+    }),
+    columnHelper.accessor('misses', {
+      id: 'misses',
+      header: () => 'Missed',
+      cell: ({ renderValue }) => renderValue(),
+    }),
+    columnHelper.accessor('coverage', {
+      id: 'percentCovered',
+      header: () => 'Coverage %',
+      cell: ({ renderValue }) => renderValue(),
+    }),
+  ]
+
+  return baseColumns
+}
+
+function FileListTable() {
+  const [sorting, setSorting] = useState([{ id: 'misses', desc: true }])
+  const ordering = useMemo(() => getOrderingDirection(sorting), [sorting])
```

Comment:
This is such a low cost function call you can probably skip wrapping it in a `useMemo` and just call it again on each re-render.